### PR TITLE
add instructions on downloading java from your package manager

### PR DESCRIPTION
**Pull request by DioEgizio**

### CODE REVIEW

**Comment by glowiak** [comment]
## Slackware

```wget https://codeberg.org/glowiak/SlackBuilds/raw/branch/master/tgz/adoptium-jdk8.tar.gz```
```tar xzvf adoptium-jdk8.tar.gz```
```(cd adoptium-jdk8 && sudo ./adoptium-jdk8.SlackBuild)```
```sudo installpkg /tmp/adoptium-jdk8*.tgz```

**Comment by Regular-Baf** [comment]
Perhaps a note stating Flatpak already includes java?

**Comment by DioEgizio** [comment]
> ## Slackware
> 
> `wget https://codeberg.org/glowiak/SlackBuilds/raw/branch/master/tgz/adoptium-jdk8.tar.gz` `tar xzvf adoptium-jdk8.tar.gz` `(cd adoptium-jdk8 && sudo ./adoptium-jdk8.SlackBuild)` `sudo installpkg /tmp/adoptium-jdk8*.tgz`

that's just java8

**Comment by glowiak** [comment]
atm

**Comment by glowiak** [comment]
the jdk17 will appear soon

**Comment by glowiak** [comment]
Added https://codeberg.org/glowiak/SlackBuilds/src/branch/master/repository/adoptium-jdk17.md

**Comment by glowiak** [comment]
so

**Comment by glowiak** [comment]
## Slackware

Java 8:

```wget https://codeberg.org/glowiak/SlackBuilds/raw/branch/master/tgz/adoptium-jdk8.tar.gz```
```tar xzvf adoptium-jdk8.tar.gz```
```(cd adoptium-jdk8 && sudo ./adoptium-jdk8.SlackBuild)```
```sudo installpkg /tmp/adoptium-jdk8*.tgz```

Java 17:

```wget https://codeberg.org/glowiak/SlackBuilds/raw/branch/master/tgz/adoptium-jdk17.tar.gz```
```tar xzvf adoptium-jdk17.tar.gz```
```(cd adoptium-jdk17 && sudo ./adoptium-jdk17.SlackBuild)```
```sudo installpkg /tmp/adoptium-jdk17*.tgz```

**Comment by Krutonium** [comment]
NixOS: `environment.systemPackages = [ pkgs.openjdk17 pkgs.openjdk8 ];`

**Comment by Krutonium** [comment]
> Is this a command?

It goes in your NixOS config to install the application.

**Comment by icelimetea** [comment]
> NixOS: environment.systemPackages = [ pkgs.openjdk17 pkgs.openjdk8 ];

PolyMC package for NixOS already depends on both of those Java versions (and can launch Minecraft using both even without explicitly adding OpenJDK to systemPackages).

File: src/wiki/getting-started/installing-java.md [hunk 1]
```diff
@@ -20,6 +20,58 @@ If you aren't sure which version you will need, **it is recommended to just inst
...
+## Flatpak
+
+The PolyMC flatpak already bundles java.
+
+## NixOS
+
+The PolyMC NixOS package should already depend on the required java versions, but if it doesn't add that to your NixOS config:
+```
+environment.systemPackages = [ pkgs.openjdk17 pkgs.openjdk8 ];
+```
```
Comment (babbaj) [inline src/wiki/getting-started/installing-java.md:70]:
```suggestion
The PolyMC NixOS package already bundles java.
```

**Comment by Krutonium** [comment]
[Technically](https://github.com/NixOS/nixpkgs/blob/dcca8d77344976752dece16c66a1ede2f379e91a/pkgs/games/polymc/default.nix#L5-L6) it specifically asks for Java 8 and whatever the latest version of the JDK is, so at the very least specifying openjdk17 is probably a good idea.

**Comment by babbaj** [comment]
Perhaps that should be changed to stay consistent with everything else

**Comment by icelimetea** [comment]
> I don't follow Java basically at all, but Java 18 could break Java 17 applications. It probably won't, but.

I think in such an event PolyMC derivation should introduce the dependency on OpenJDK 17 explicitly, as it already does with OpenJDK 8 for the same purposes.

File: src/wiki/getting-started/installing-java.md [hunk 2]
```diff
@@ -20,6 +20,55 @@ If you aren't sure which version you will need, **it is recommended to just inst
...
+sudo apt install openjdk-17-jre openjdk-8-jre
+```
+
+## Debian
+
+```
+sudo apt install openjdk-17-jre 
+```
+
+Java 8 is not available from Debian repos for unknown reasons, but you can [get a deb from Azul](https://www.azul.com/downloads/?version=java-8-lts&os=debian&package=jre&show-old-builds=true)
```
Comment (icelimetea) [inline src/wiki/getting-started/installing-java.md:59]: Looks like it actually is available, but only on oldoldstable and sid branches:
image

Comment (icelimetea) [reply src/wiki/getting-started/installing-java.md:59]: https://packages.debian.org/search?keywords=openjdk-8-jre&searchon=names&suite=all&section=all

Comment (DioEgizio) [reply src/wiki/getting-started/installing-java.md:59]: well, you shouldn't use oldoldstable (PolyMC won't work on it anyways) and most people aren't going to use those versions

**Review by Scrumplex — APPROVED** [review]

**Pull request merged**